src/CMakeLists.txt

```diff
@@ -399,31 +404,148 @@ if(${NUPIC_IWYU})
+                         COMMAND ${CMAKE_AR} -x ${LIB_STATIC_Z_LOC}
+                         COMMAND ${CMAKE_AR} rcs ${src_lib_static_nupiccore_combined} *.o
+                         DEPENDS ${src_nupiccore_combined_deps})
+    endif()
+  else()
+    set(src_lib_static_nupiccore_combined "${PROJECT_BINARY_DIR}/nupic_core.lib")
+    add_custom_command(OUTPUT ${src_lib_static_nupiccore_combined}
+                       COMMAND lib.exe /OUT:${src_lib_static_nupiccore_combined}
+                               $<TARGET_FILE:${src_lib_static_nupiccore_solo}>
+                               ${LIB_STATIC_YAML_CPP_LOC}
+                               ${LIB_STATIC_YAML_LOC}
+                               ${LIB_STATIC_APRUTIL1_LOC}
+                               ${LIB_STATIC_APR1_LOC}
+                               ${LIB_STATIC_Z_LOC}
+                       DEPENDS ${src_nupiccore_combined_deps}
+                       VERBATIM)
+  endif()
+
+
+  # Wrap nupic.core combined lib output in custom target to avoid conflict from
+  # other dependent targets that may execute in parallel (per add_custom_command
+  # documentation). NOTE Other targets MUST use LibStaticNupicCoreCombinedTarget
+  # as a dependency instead of ${src_lib_static_nupiccore_combined}
+  add_custom_target(LibStaticNupicCoreCombinedTarget
+                    DEPENDS ${src_lib_static_nupiccore_combined})
+else()
+
+  #
+  # Combine nupic_core_solo static lib with our external static libraries into
+  # the nupic_core static library
+  #
+
+  set(src_lib_static_nupiccore_combined nupic_core)
+
+  include(src/NupicLibraryUtils) # for MERGE_STATIC_LIBRARIES
+
+  set(src_external_static_lib_deps
+      Apr1StaticLib
+      AprUtil1StaticLib
+      ZStaticLib
+      YamlStaticLib
+      YamlCppStaticLib)
+
+  set(src_external_static_libs
+      ${LIB_STATIC_YAML_CPP_LOC}
+      ${LIB_STATIC_YAML_LOC}
+      ${LIB_STATIC_APR1_LOC}
+      ${LIB_STATIC_APRUTIL1_LOC}
+      ${LIB_STATIC_Z_LOC})
+
+  # Add capnproto static libs when building nupic.core static lib without python
+  # extensions
+  if (NOT ${NUPIC_BUILD_PYEXT_MODULES})
+    list(APPEND src_external_static_lib_deps CapnProto)
+    list(APPEND src_external_static_libs ${CAPNP_LINK_LIBRARIES})
+  endif() # ${NUPIC_BUILD_PYEXT_MODULES}
+
+  set(src_combined_nupiccore_source_archives
+      ${src_lib_static_nupiccore_solo}
+      ${src_external_static_libs})
+
+  # Create a top-level library target for the combined static lib
+  merge_static_libraries(${src_lib_static_nupiccore_combined}
+                         "${src_combined_nupiccore_source_archives}")
+  add_dependencies(${src_lib_static_nupiccore_combined}
+                   ${src_external_static_libs})
+endif()
+
+
+#
+# Build tests of the nupic_core "combined" static library
+#
+
 # Common libs for test executables
 set(src_common_test_exe_libs
-    ${src_lib_static_nupiccore_solo}
-    ${src_common_libs}
+    ${src_lib_static_nupiccore_combined}
```

Comment:
The tests are now linked with the output artifact nupic_core "combined" static lib instead of the intermediate nupic_core_solo static lib.